state-sync/data-streaming-service/src/data_stream.rs

```diff
@@ -303,8 +314,9 @@ impl<T: AptosDataClientInterface + Send + Clone + 'static> DataStream<T> {
             );
         }
 
-        // Update the counters for the pending response queue
-        metrics::set_pending_data_responses(self.get_sent_data_requests()?.len());
+        // Update the counters for the complete and pending responses
+        metrics::set_complete_pending_data_responses(num_complete_pending_requests);
+        metrics::set_pending_data_responses(self.get_num_pending_data_requests()?);
```

Comment:
is this different than the value we fetched at the top of the function?

Reply:
Yeah, it will have updated according to how many new requests were put in to the queue (between the two calls) 😄